$NetBSD: patch-Makefile,v 1.5 2026/02/24 23:09:02 thor Exp $

Use universal LNCMD that works for all library naming schemes (just
skipping linking if the file is already named like the link).

--- Makefile.orig	2026-01-15 22:57:26.000000000 +0000
+++ Makefile
@@ -1,9 +1,5 @@
 TOPDIR	= .
 include ./Makefile.system
-LNCMD = ln -fs
-ifeq ($(FIXED_LIBNAME), 1)
-LNCMD = true
-endif
 
 BLASDIRS = interface driver/level2 driver/level3 driver/others
 
@@ -95,7 +91,7 @@ ifeq ($(NOFORTRAN), $(filter 0,$(NOFORTR
 endif
 
 ifeq ($(OSNAME), WINNT)
-	@-$(LNCMD) $(LIBNAME) $(LIBPREFIX).$(LIBSUFFIX)
+	@-$(call LNCMD,$(LIBNAME),$(LIBPREFIX).$(LIBSUFFIX))
 endif
 
 ifneq ($(OSNAME), AIX)
@@ -147,17 +143,17 @@ shared : libs netlib $(RELA)
 ifneq ($(NO_SHARED), 1)
 ifeq ($(OSNAME), $(filter $(OSNAME),Linux SunOS Android Haiku FreeBSD DragonFly))
 	@$(MAKE) -C exports so
-	@$(LNCMD) $(LIBSONAME) $(LIBPREFIX).so
-	@$(LNCMD) $(LIBSONAME) $(LIBPREFIX).so.$(MAJOR_VERSION)
+	@$(call LNCMD,$(LIBSONAME),$(LIBPREFIX).so)
+	@$(call LNCMD,$(LIBSONAME),$(LIBPREFIX).so.$(MAJOR_VERSION))
 endif
 ifeq ($(OSNAME), $(filter $(OSNAME),OpenBSD NetBSD))
 	@$(MAKE) -C exports so
-	@$(LNCMD) $(LIBSONAME) $(LIBPREFIX).so
+	@$(call LNCMD,$(LIBSONAME),$(LIBPREFIX).so)
 endif
 ifeq ($(OSNAME), Darwin)
 	@$(MAKE) -C exports dyn
-	@$(LNCMD) $(LIBDYNNAME) $(LIBPREFIX).dylib
-	@$(LNCMD) $(LIBDYNNAME) $(LIBPREFIX).$(MAJOR_VERSION).dylib
+	@$(call LNCMD,$(LIBDYNNAME),$(LIBPREFIX).dylib)
+	@$(call LNCMD,$(LIBDYNNAME),$(LIBPREFIX).$(MAJOR_VERSION).dylib)
 endif
 ifeq ($(OSNAME), WINNT)
 	@$(MAKE) -C exports dll
@@ -245,13 +241,13 @@ ifeq ($(INTERFACE64),1)
 endif
 	@echo THELIBNAME=$(LIBNAME) >>  Makefile.conf_last
 	@echo THELIBSONAME=$(LIBSONAME) >>  Makefile.conf_last
-	@-$(LNCMD) $(LIBNAME) $(LIBPREFIX).$(LIBSUFFIX)
+	@-$(call LNCMD,$(LIBNAME),$(LIBPREFIX).$(LIBSUFFIX))
 	@touch lib.grd
 
 prof : prof_blas prof_lapack
 
 prof_blas :
-	$(LNCMD) $(LIBNAME_P) $(LIBPREFIX)_p.$(LIBSUFFIX)
+	$(call LNCMD,$(LIBNAME_P),$(LIBPREFIX)_p.$(LIBSUFFIX))
 	for d in $(SUBDIRS) ; \
 	do if test -d $$d; then \
 	  $(MAKE) -C $$d prof || exit 1 ; \
@@ -262,7 +258,7 @@ ifeq ($(DYNAMIC_ARCH), 1)
 endif
 
 blas :
-	$(LNCMD) $(LIBNAME) $(LIBPREFIX).$(LIBSUFFIX)
+	$(call LNCMD,$(LIBNAME),$(LIBPREFIX).$(LIBSUFFIX))
 	for d in $(BLASDIRS) ; \
 	do if test -d $$d; then \
 	  $(MAKE) -C $$d libs || exit 1 ; \
@@ -270,7 +266,7 @@ blas :
 	done
 
 hpl :
-	$(LNCMD) $(LIBNAME) $(LIBPREFIX).$(LIBSUFFIX)
+	$(call LNCMD,$(LIBNAME),$(LIBPREFIX).$(LIBSUFFIX))
 	for d in $(BLASDIRS) ../laswp exports ; \
 	do if test -d $$d; then \
 	  $(MAKE) -C $$d $(@F) || exit 1 ; \
@@ -284,7 +280,7 @@ ifeq ($(DYNAMIC_ARCH), 1)
 endif
 
 hpl_p :
-	$(LNCMD) $(LIBNAME_P) $(LIBPREFIX)_p.$(LIBSUFFIX)
+	$(call LNCMD,$(LIBNAME_P),$(LIBPREFIX)_p.$(LIBSUFFIX))
 	for d in $(SUBDIRS) ../laswp exports ; \
 	do if test -d $$d; then \
 	  $(MAKE) -C $$d $(@F) || exit 1 ; \
